client/src/app/pages/external/jira/instance-form.tsx

```diff
@@ -0,0 +1,292 @@
+  });
+
+  const {
+    handleSubmit,
+    formState: { isSubmitting, isValidating, isValid, isDirty },
+    getValues,
+    setValue,
+    control,
+    watch,
+  } = useForm<FormValues>({
+    defaultValues: {
+      name: instance?.name || "",
+      url: instance?.url || "",
+      id: instance?.id || 0,
+      kind: instance?.kind,
+      credentialName: instance?.identity?.name || "",
+    },
+    resolver: yupResolver(validationSchema),
+    mode: "onChange",
+  });
+
+  const values = getValues();
+
+  const [isKeyFileRejected, setIsKeyFileRejected] = useState(false);
+  const [isSettingsFileRejected, setIsSettingsFileRejected] = useState(false);
+
+  const watchAllFields = watch();
+
+  const kindOptions: OptionWithValue<JiraKind>[] = [
+    {
+      value: "jira-cloud",
+      toString: () => "Jira Cloud",
+    },
+    {
+      value: "jira-server",
+      toString: () => "Jira Server",
+    },
+    {
+      value: "jira-datacenter",
+      toString: () => "Jira Datacenter",
+    },
+  ];
+
+  const identityOptions = identities
+    .filter((identity) => identity.kind === "jira")
+    .map((identity) => {
+      return {
+        value: identity.name,
+        toString: () => identity.name,
+      };
+    });
+
+  return (
+    <Form onSubmit={handleSubmit(onSubmit)}>
+      {axiosError && (
+        <Alert
+          variant="danger"
+          isInline
+          title={getAxiosErrorMessage(axiosError)}
+        />
+      )}
+      <HookFormPFTextInput
+        control={control}
+        name="name"
+        label="Instance name"
+        fieldId="name"
+        isRequired
+      />
+      <HookFormPFTextInput
+        control={control}
+        name="url"
+        label="URL"
+        fieldId="url"
+        isRequired
+      />
+
+      <HookFormPFGroupController
+        control={control}
+        name="kind"
+        label="Instance type"
+        fieldId="type-select"
+        isRequired
+        renderInput={({ field: { value, name } }) => (
+          <SimpleSelect
+            id="type-select"
+            toggleId="type-select-toggle"
+            placeholderText={t("composed.selectMany", {
+              what: t("terms.instanceType").toLowerCase(),
+            })}
+            toggleAriaLabel="Type select dropdown toggle"
+            aria-label={name}
+            value={value ? toOptionLike(value, kindOptions) : undefined}
+            options={kindOptions}
+            onChange={(selection) => {
+              const selectionValue = selection as OptionWithValue<JiraKind>;
+              setValue(name, selectionValue.value);
+            }}
+          />
+        )}
+      />
+      <HookFormPFGroupController
+        control={control}
+        name="credentialName"
+        label={t("terms.credentials")}
+        fieldId="Credentials"
+        isRequired
+        renderInput={({ field: { value, name, onChange } }) => (
+          <SimpleSelect
+            id="credentials-select"
```

Comment:
Need to add: 

```
  maxHeight={DEFAULT_SELECT_MAX_HEIGHT}
  variant="typeahead"
```

This adds a default no results found when no identities are available. Also the max height helps with overflow in the case of a large number of options. Check the application-identity-form.tsx identity dropdown for an example.